src/PrestaShopBundle/Controller/ArgumentResolver/SearchParametersResolver.php

```diff
@@ -0,0 +1,153 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Controller\ArgumentResolver;
+
+use PrestaShop\PrestaShop\Core\Search\ControllerAction;
+use PrestaShop\PrestaShop\Core\Search\SearchParameters;
+use PrestaShopBundle\Entity\Repository\AdminFilterRepository;
+use PrestaShopBundle\Security\Admin\Employee;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\HttpFoundation\Request;
+use PrestaShop\PrestaShop\Core\Search\Filters;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+
+/**
+ * If an action inject instance of Filters, this class is responsible of
+ * creating it from available sources.
+ */
+class SearchParametersResolver implements ArgumentValueResolverInterface
+{
+    /**
+     * @var SearchParameters
+     */
+    private $searchParameters;
+
+    /**
+     * @var AdminFilterRepository
+     */
+    private $adminFilterRepository;
+
+    /**
+     * @var Employee|void
+     */
+    private $employee;
+
+    /**
+     * @var int
+     */
+    private $shopId;
+
+    /**
+     * SearchParametersResolver constructor.
+     *
+     * @param SearchParameters      $searchParameters
+     * @param TokenStorageInterface $tokenStorage
+     * @param AdminFilterRepository $adminFilterRepository
+     * @param int                   $shopId           The Shop id
+     */
+    public function __construct(
+        SearchParameters $searchParameters,
+        TokenStorageInterface $tokenStorage,
+        AdminFilterRepository $adminFilterRepository,
```

Comment:
use the interface or EntityRepository at least

Reply:
thinking about it, not possible except for creating a really specific interface: it's easier to override the service: abort ^_^